MM-28227 - Add mmctl role reset command (#4162)
mattermost/docs: source/administration/mmctl-cli-tool.rst

Management of permissions and roles.

Child Commands
  -  `mmctl permissions add`_ - Add permissions to a role
  -  `mmctl permissions role assign`_ - Assign users to role
  -  `mmctl permissions remove`_ - Remove permissions from a role

>   -  `mmctl permissions reset`_ - Reset default permissions for role

Reviewer: ```suggestion
  -  `mmctl permissions reset`_ - Reset default permissions for a role
```

>   -  `mmctl permissions role show`_ - Show the role information

Reviewer 2: ```suggestion
  -  `mmctl permissions show`_ - Show the role information
```

>   -  `mmctl permissions role unassign`_ - Unassign users from role

Reviewer: ```suggestion
  -  `mmctl permissions role unassign`_ - Unassign users from a role
```

   --local                       allows communicating with the server through a unix socket
   --strict                      will only run commands if the mmctl version matches the server one

mmctl permissions role assign
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

**Description**

> Assign users to a role by username (only available in E10 and E20).

Reviewer: ```suggestion
Assign users to a role by username (only available in Enterprise Edition E10 and E20).
```

**Format**

.. code-block:: sh

  mmctl permissions role assign [role_name] [username...] [flags]

**Examples**

.. code-block:: sh

>     # Assign users with usernames 'john.doe' and 'jane.doe' to the role named 'system_admin'.

Reviewer: I really like the addition of this context. One suggestion I have is to remove it from the code block and have it as normal text so it's more visually obvious. Thoughts?

Author: Content in code blocks are easy to copy and paste directly. My recommendation is to limit code blocks to only code that copies and pastes successfully for the user, and to split out great context like this as normal text.

Reviewer: 5/5 on limiting code blocks to that and splitting out the context and code. I do see the other instances of this and I think it's early enough to fix them and then going forward separate context and code?

Reviewer: Based on above, let's leave this as-is being that it matches the current format and then look at changing it post-release in a new PR.

Author: Great! This update would be a great addition to the Style Guide. Once included, we can perform a sweep of the current docs to adjust and align to the Style Guide. Thoughts?

Reviewer: I'm 5/5 on that! Will add it to our list. :)

   --local                       allows communicating with the server through a unix socket
   --strict                      will only run commands if the mmctl version matches the server one

mmctl permissions reset
^^^^^^^^^^^^^^^^^^^^^^^

**Description**

> Reset the given role's permissions to the set that was originally released with (only available in E10 and E20).

Reviewer: ```suggestion
Reset the given role's permissions to the default settings and overwrite custom settings (only available in Enterprise Edition E10 and E20).
```

Reviewer: @mgdelacroix, is my suggestion correct?

   --local                       allows communicating with the server through a unix socket
   --strict                      will only run commands if the mmctl version matches the server one

mmctl permissions role unassign
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

**Description**

> Unassign users from a role by username (only available in E10 and E20).

Reviewer: ```suggestion
Unassign users from a role by username (only available in Enterprise Edition E10 and E20).
```

**Format**

.. code-block:: sh

  mmctl permissions role unassign [role_name] [username...] [flags]

**Examples**

.. code-block:: sh

>     # Unassign users with usernames 'john.doe' and 'jane.doe' from the role named 'system_admin'.

Reviewer: Same comment as above. :)

Reviewer: Same retraction as above. :)
